marlowe-playground-client/src/Simulation/BottomPanel.purs

```diff
@@ -318,15 +331,43 @@ panelContents state MarloweErrorsView =
     ]
     content
   where
-  errors = state ^. (_marloweState <<< _Head <<< _editorErrors)
+  errors = state ^. (_marloweState <<< _Head <<< _editorErrors <<< to (map formatError))
 
-  content = if Array.null errors then [ pre [ class_ (ClassName "error-content") ] [ text "No errors" ] ] else map renderError errors
+  content =
+    if Array.null errors then
+      [ pre [ class_ (ClassName "error-content") ] [ text "No errors" ] ]
+    else
+      [ div [ classes [ ClassName "error-headers", ClassName "error-row" ] ]
+          [ div [] [ text "Description" ]
+          , div [] [ text "Line Number" ]
+          ]
+      , ul [] (map renderError errors)
+      ]
 
   renderError error =
-    pre [ class_ (ClassName "error-content") ]
-      [ a [ onClick $ const $ Just $ MarloweMoveToPosition error.startLineNumber error.startColumn ]
-          [ text error.message ]
-      ]
+    li [ classes [ ClassName "error-content", ClassName "error-row", ClassName "flex-wrap" ] ]
+      ( [ a [ onClick $ const $ Just $ ShowErrorDetail (state ^. (_showErrorDetail <<< to not)) ]
+            [ text error.firstLine ]
+        , a [ onClick $ const $ Just $ MarloweMoveToPosition error.startLineNumber error.startColumn ]
+            [ text $ show error.startLineNumber ]
+        ]
+          <> if (state ^. _showErrorDetail) then
+              [ pre [ class_ (ClassName "error-content") ] [ text error.restLines ] ]
+            else
+              []
+      )
+
+  formatError { message, startColumn, startLineNumber } =
+    let
+      lines' = lines message
+
+      firstLine /\ restLines =
+        if (take 12 $ fromMaybe mempty $ head lines') == "Syntax error" then
+          "Syntax error" /\ (unlines $ fromMaybe [] $ tail $ drop 1 lines')
```

Comment:
Just checking this is deliberate - you're truncating the first line if it did begin with "Syntax error".

Reply:
yes, I have little control over the text of the errors but I want to structure them a little bit so I can have an overview, a location and the detailed description.

Reply:
OFC I could always do some disgusting javascript prototype overriding weirdness I suppose but...no